### PR TITLE
Support for Universal Module Definition (UMD)

### LINKED ISSUE #174: Annyang not working in Electron apps
Hi,

I tried using Annyang for some speech detection in an Electron app and it doesn't seem to be working. I receive a typeError(not a function) whenever I try to use any of its functions. Any reason why this would be happening since Electron does run in a Chromium environment.

To overcome the CommonJS issues I'm loading annyang in the app using 

`<script>window.annyang = require('/path to file')</script>`

However when I access annyang in the dev console it shows that it's a null object.

### PR DESCRIPTION
**Pull request by v12**
This pull request adds [Universal Module Definition](https://github.com/umdjs/umd) support, allowing **annyang** to be used as a module.

## Motivation and Context
First of all, support for UMD is vital nowadays because many developers prefer to use modules. It's 2016 :tada: and we use [Webpack](https://github.com/webpack/webpack)/[Browserify](https://github.com/substack/node-browserify)/[RequireJS](https://github.com/requirejs/requirejs)/etc :tophat: in order to speed up development using modules. However, there are two major patterns that are used by developers when defining their modules - AMD and CommonJS. It'd be disrespectful to support single approach withdrawing a lot of potential users by not supporting another one.

Secondly, it resolves #174.

## How Has This Been Tested?

Because there are no changes in API, test suite was not changed.

Furthermore, support for UMD is done in backward compatible manner, so there is no need for major version bump. `annyang` is still available as a property of global `window` object. 

The only change to tests is the use of UMD when initialising the suite (updated for Corti as well). However, in order to test the module when it's used as a module, test suite is ran once again but with the use of `grunt-template-jasmine-requirejs`

JSHint configuration had to be updated in order to allow `define` to be a global variable.

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by v12** [comment]
@TalAter is there anything that prevents this pull request from being merged? :wink:

File: annyang.js [hunk 1]
```diff
@@ -3,8 +3,19 @@
...
+  if (typeof define === 'function' && define.amd) { // AMD
+    define([], function () {
+      return (root.annyang = factory(root, undefined));
+    });
+  } else if (typeof module === 'object' && module.exports) { // CommonJS
+    module.exports = factory(root, undefined);
+  } else { // Browser globals
+    root.annyang = factory(root, undefined);
+  }
+}(typeof window !== 'undefined' ? window : this, function (root, undefined) {
```
Comment (TalAter) [inline annyang.js]: Why do you need to check for the existence of window here?
I am guessing this comes in handy if using AMD or CommonJS in the browser/electron, but I would like to understand the code completely.

**Comment by TalAter** [comment]
Hi @v12 
Thanks for the outstanding contribution. I think this is the most thought out pull request I've ever seen. :+1: 

My apologies for taking so long to respond. As you can imagine, going over a change this big requires a lot of consideration and attention... I had to set aside a whole day just to go over everything.

I have a number of questions which I am adding to this issue as I find them.

Thank you, and happy 2016 :wink:

**Comment by TalAter** [comment]
Merged.
Truly great work! Thank you.

**Pull request merged**